Description: ignore SIGPIPE in more tests
 These are the tests which failed (killed with SIGPIPE, Wstat=13) when
 run continuously on a CPU-loaded system.
Author: Damyan Ivanov <dmn@debian.org>

--- a/t/compatibility.t
+++ b/t/compatibility.t
@@ -14,6 +14,7 @@ Test::More->builder->use_numbers(0);
 Test::More->builder->no_ending(1);
 
 $SIG{'CHLD'} = "IGNORE";
+$SIG{'PIPE'} = "IGNORE";
 
 IO::Socket::SSL::context_init(SSL_verify_mode => 0x01);
 
--- a/t/verify_hostname.t
+++ b/t/verify_hostname.t
@@ -14,6 +14,7 @@ my $can_idn  = eval { require Encode } &
     || eval { require URI; URI->VERSION(1.50) }
 );
 
+$SIG{'PIPE'} = "IGNORE";
 $|=1;
 my $max = 42;
 $max+=3 if $can_idn;
--- a/t/startssl.t
+++ b/t/startssl.t
@@ -8,6 +8,8 @@ use IO::Socket::SSL;
 use IO::Select;
 do './testlib.pl' || do './t/testlib.pl' || die "no testlib";
 
+$SIG{'PIPE'} = "IGNORE";
+
 $|=1;
 print "1..21\n";
 
--- a/t/dhe.t
+++ b/t/dhe.t
@@ -11,6 +11,7 @@ use Socket;
 use IO::Socket::SSL;
 do './testlib.pl' || do './t/testlib.pl' || die "no testlib";
 
+$SIG{'PIPE'} = "IGNORE";
 $|=1;
 print "1..3\n";
 
--- a/t/ecdhe.t
+++ b/t/ecdhe.t
@@ -14,6 +14,7 @@ if ( ! IO::Socket::SSL->can_ecdh ) {
     exit
 }
 
+$SIG{'PIPE'} = "IGNORE";
 $|=1;
 print "1..4\n";
 
--- a/t/alpn.t
+++ b/t/alpn.t
@@ -17,6 +17,7 @@ if ( ! IO::Socket::SSL->can_alpn ) {
     exit;
 }
 
+$SIG{'PIPE'} = "IGNORE";
 print "1..5\n";
 
 # first create simple ssl-server
--- a/t/io-socket-ip.t
+++ b/t/io-socket-ip.t
@@ -29,6 +29,8 @@ unless( IO::Socket::SSL->CAN_IPV6 eq "IO
     exit
 }
 
+$SIG{'PIPE'} = "IGNORE";
+
 my $addr = '::1';
 # check if we can use ::1, e.g. if the computer has IPv6 enabled
 if ( ! IO::Socket::IP->new(
--- a/t/io-socket-inet6.t
+++ b/t/io-socket-inet6.t
@@ -37,6 +37,8 @@ unless( IO::Socket::SSL->CAN_IPV6 eq "IO
     exit
 }
 
+$SIG{'PIPE'} = "IGNORE";
+
 my $addr = '::1';
 # check if we can use ::1, e.g. if the computer has IPv6 enabled
 if ( ! IO::Socket::INET6->new(
--- a/t/public_suffix_ssl.t
+++ b/t/public_suffix_ssl.t
@@ -34,6 +34,7 @@ my @tests = qw(
     ok:www.foo.nodomain|*.foo.nodomain
 );
 
+$SIG{'PIPE'} = "IGNORE";
 $|=1;
 plan tests => 0+@tests;
 
--- a/t/sessions.t
+++ b/t/sessions.t
@@ -9,6 +9,7 @@ use Socket;
 use IO::Socket::SSL;
 do './testlib.pl' || do './t/testlib.pl' || die "no testlib";
 
+$SIG{'PIPE'} = "IGNORE";
 $|=1;
 my $numtests = 35;
 print "1..$numtests\n";
--- a/t/verify_fingerprint.t
+++ b/t/verify_fingerprint.t
@@ -8,6 +8,8 @@ do './testlib.pl' || do './t/testlib.pl'
 
 plan tests => 13;
 
+$SIG{'PIPE'} = "IGNORE";
+
 my ($ca1,$cakey1) = CERT_create( CA => 1, subject => { CN => 'ca1' });
 my ($cert1,$key1) = CERT_create( 
     subject => { CN => 'cert1' },
